### PR TITLE
Fix GHA workflows naming & Run ITs if UTs fail on coverage

### PR DESCRIPTION
**Pull request by tejaswini-imply**
Currently, there is no way to run ITs if unit-tests fail on coverage. This PR allows Revised, Standard ITs to run when unit-tests fail on coverage errors, still failing the workflow. This PR also fixes existing GHA workflow naming.

screenshot_2023-04-24_at_6 30 54_pm

### CODE REVIEW

File: .github/scripts/unit_tests_script.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@ MAVEN_OPTS='-Xmx2500m' ${MVN} test -pl ${MAVEN_PROJECTS} \
 ${MAVEN_SKIP} -Ddruid.generic.useDefaultValueForNull=${DRUID_USE_DEFAULT_VALUE_FOR_NULL}
 sh -c "dmesg | egrep -i '(oom|out of memory|kill process|killed).*' -C 1 || exit 0"
 free -m
-${MVN} -pl ${MAVEN_PROJECTS} jacoco:report
+${MVN} -pl ${MAVEN_PROJECTS} jacoco:report || { echo echo "coverage_failure=false" >> "$GITHUB_ENV" && false; }
```
Comment (abhishekagarwal87) [inline .github/scripts/unit_tests_script.sh:27]: echo echo?

File: .github/scripts/unit_tests_script.sh [hunk 2]
```diff
@@ -70,7 +70,7 @@ then
   --log-template "totals-complete" \
   --log-template "errors" \
   -- ||
-  { printf "\n\n****FAILED****\nDiff code coverage check failed. To view coverage report, run 'mvn clean test jacoco:report' and open 'target/site/jacoco/index.html'\nFor more details on how to run code coverage locally, follow instructions here - https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md#running-code-coverage-locally\n\n" && false; }
+  { printf "\n\n****FAILED****\nDiff code coverage check failed. To view coverage report, run 'mvn clean test jacoco:report' and open 'target/site/jacoco/index.html'\nFor more details on how to run code coverage locally, follow instructions here - https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md#running-code-coverage-locally\n\n" && printf "test print\n" && echo "coverage_failure=true" >> "$GITHUB_ENV" && printf "test print2\n" && false; }
```
Comment (abhishekagarwal87) [inline .github/scripts/unit_tests_script.sh:73]: lot of test printing here.

File: .github/workflows/cron-job-its.yml [hunk 3]
```diff
@@ -13,7 +13,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-name: Cron Job ITs
```
Comment (abhishekagarwal87) [inline .github/workflows/cron-job-its.yml:16]: whats the reason for these changes?

File: .github/workflows/reusable-standard-its.yml [hunk 4]
```diff
@@ -89,7 +89,7 @@ jobs:
           ${MVN} verify -pl integration-tests -P integration-tests ${{ inputs.testing_groups }} -Djvm.runtime=${{ inputs.runtime_jdk }} -Dit.indexer=${{ inputs.use_indexer }} ${MAVEN_SKIP} -Doverride.config.path=${{ inputs.override_config_path }}
 
       - name: Debug IT
-        if: steps.run-it.conclusion == 'failure'
+        if: ${{ failure() && steps.run-it.conclusion == 'failure' }}
```
Comment (abhishekagarwal87) [inline .github/workflows/reusable-standard-its.yml:92]: what's the reason behind this change?

File: .github/workflows/unit-tests.yml [hunk 5]
```diff
@@ -25,6 +26,16 @@ on:
         type: boolean
         default: true
         description: 'For SQL compatibility'
+    outputs:
+      continue_tests:
+        description: 'Flag to decide if phase 2 tests need to run incase coverage issue failures'
```
Comment (abhishekagarwal87) [inline .github/workflows/unit-tests.yml:31]: we are not using phase 1 / phase 2 terminology in GHA.

File: .github/workflows/unit-tests.yml [hunk 6]
```diff
@@ -50,7 +61,7 @@ jobs:
       sql_compatibility: ${{ inputs.sql_compatibility }}
       module: server
       maven_projects: 'server'
-
+#
```
Comment (abhishekagarwal87) [inline .github/workflows/unit-tests.yml:64]: leftover change?

File: .github/workflows/cron-job-its.yml (hunk 3 above)
Comment (tejaswini-imply) [reply .github/workflows/cron-job-its.yml:16]: Just to make naming consistent across workflows.

File: .github/workflows/reusable-standard-its.yml (hunk 4 above)
Comment (tejaswini-imply) [reply .github/workflows/reusable-standard-its.yml:92]: https://docs.github.com/en/actions/learn-github-actions/expressions#failure-with-conditions Just to be explicit.

**Comment by tejaswini-imply** [comment]
Thanks @abhishekagarwal87 for the review, I have addressed your comments in latest commit.

**Review by abhishekagarwal87 — APPROVED** [review]
